src/Microsoft.ML.Core/Utilities/MathUtils.cs

```diff
@@ -871,5 +871,16 @@ public static double Cos(double a)
             var res = Math.Cos(a);
             return Math.Abs(res) > 1 ? double.NaN : res;
         }
+
+        /// <summary>
+        /// Returns the smallest integral value that is greater than or equal to the result of the division.
+        /// </summary>
+        /// <param name="numerator">Number to be divided.</param>
+        /// <param name="denomenator">Number with which to divide the numerator.</param>
+        /// <returns></returns>
+        public static long DivisionCeiling(long numerator, long denomenator)
+        {
+            return (numerator + denomenator - 1) / denomenator;
```

Comment:
>(numerator + denomenator [](start = 19, length = 24)

should this be checked? what if numerator + denumerator > long?  Will we get a hidden overflow?